rpcs3/evdev_joystick_handler.cpp

```diff
@@ -13,560 +17,697 @@
+		if (button_code < 0 || value < 0)
+			continue;
+
+		bool is_button_or_trigger = evt.type == EV_KEY || button_code == ABS_Z || button_code == ABS_RZ;
+
+		// Translate any corresponding keycodes to our normal DS3 buttons and triggers
+		for (int i = 0; i < static_cast<int>(pad->m_buttons.size() - 1); i++) // skip reserved button
+		{
+			if (pad->m_buttons[i].m_keyCode != button_code)
+				continue;
+
+			// Be careful to handle mapped axis specially
+			if (evt.type == EV_ABS)
+			{
+				// get axis direction and skip on error or set to 0 if the stick/hat is actually pointing to the other direction.
+				// maybe mimic on error, needs investigation. FindAxisDirection should ideally never return -1 anyway
+				int direction = FindAxisDirection(axis_orientations, i);
+				if (direction < 0)
+				{
+					LOG_ERROR(HLE, "FindAxisDirection = %d, Button Nr.%d, value = %d", direction, i, value);
+					continue;
+				}
+				else if (direction != (is_negative ? 1 : 0))
+				{
+					pad->m_buttons[i].m_value = 0;
+					pad->m_buttons[i].m_pressed = 0;
+					continue;
+				}
+			}
+
+			pad->m_buttons[i].m_value = static_cast<u16>(value);
+			TranslateButtonPress(button_code, pad->m_buttons[i].m_pressed, pad->m_buttons[i].m_value);
+		}
+
+		// Translate any corresponding keycodes to our two sticks. (ignoring thresholds for now)
+		for (int idx = 0; idx < static_cast<int>(pad->m_sticks.size()); idx++)
+		{
+			bool pressed_min = false, pressed_max = false;
+			u16 val_min = 0, val_max = 0;
+
+			// m_keyCodeMin is the mapped key for left or down
+			if (pad->m_sticks[idx].m_keyCodeMin == button_code)
+			{
+				bool is_direction_min = false;
+
+				if (!is_button_or_trigger && evt.type == EV_ABS)
+				{
+					int index = BUTTON_COUNT + (idx * 2);
+					int min_direction = FindAxisDirection(axis_orientations, index);
+					if (min_direction < 0)
+					{
+						LOG_ERROR(HLE, "keyCodeMin FindAxisDirection = %d, Axis Nr.%d, Button Nr.%d, value = %d", min_direction, idx, index, value);
+					}
+					else
+					{
+						is_direction_min = is_negative == (min_direction == 1 ? true : false);
+					}
+				}
+
+				if (is_button_or_trigger || is_direction_min)
+				{
+					val_min = value;
+					TranslateButtonPress(button_code, pressed_min, val_min, true);
+				}
+				else // set to 0 to avoid remnant counter axis values
+					val_min = 0;
+			}
+
+			// m_keyCodeMax is the mapped key for right or up
+			if (pad->m_sticks[idx].m_keyCodeMax == button_code)
+			{
+				bool is_direction_max = false;
+
+				if (!is_button_or_trigger && evt.type == EV_ABS)
+				{
+					int index = BUTTON_COUNT + (idx * 2) + 1;
+					int max_direction = FindAxisDirection(axis_orientations, index);
+					if (max_direction < 0)
+					{
+						LOG_ERROR(HLE, "keyCodeMax FindAxisDirection = %d, Axis Nr.%d, Button Nr.%d, value = %d", max_direction, idx, index, value);
+					}
+					else
+					{
+						is_direction_max = is_negative == (max_direction == 1 ? true : false);
```

Comment:
is_direction_max = is_negative == (max_direction == 1 ? false : true);

Reply:
With these changes, it's completely coherent on my system: when I record an axis: it's correct. When I move my stick: it's correct.
If there's no config file, then my xbox 360 controller works fine with the default settings.

Reply:
the code is correct. the issue you are referring to was a faulty index